python_modules/dagster/dagster/_core/definitions/partition.py

```diff
@@ -636,6 +636,18 @@ def get_partition_keys(
                 partitions_def_name=self._validated_name()
             )
 
+    def get_serializable_unique_identifier(
+        self, dynamic_partitions_store: Optional[DynamicPartitionsStore] = None
+    ) -> str:
+        if not dynamic_partitions_store:
+            return super().get_serializable_unique_identifier(
+                dynamic_partitions_store=dynamic_partitions_store
+            )
+        else:
+            return dynamic_partitions_store.get_dynamic_partitions_definition_id(
+                self._validated_name()
+            )
```

Comment:
i am uncertain about the fact that this changes behavior if somebody forgets to pass in the dynamic partititons store. Should there be some kind of assert if you try to use this on a dynamic partitions definition and have forgotten to include that? or is that already covered elsewhere?

Reply:
yeah, it's not obvious here, but this is only overridden in the `DynamicPartitionsDefinition` case.  The `super` call would eventually hit the `get_partition_keys` call, which would raise a runtime error.  I'll refactor to make this more clear.